exchanges/ftx/ftx_wrapper.go

```diff
@@ -0,0 +1,833 @@
+		resp.Last = markets.Result[x].Last
+		resp.Bid = markets.Result[x].Bid
+		resp.Ask = markets.Result[x].Ask
+		resp.LastUpdated = time.Now()
+		err = ticker.ProcessTicker(f.Name, &resp, assetType)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return ticker.GetTicker(f.Name, p, assetType)
+}
+
+// FetchTicker returns the ticker for a currency pair
+func (f *FTX) FetchTicker(p currency.Pair, assetType asset.Item) (*ticker.Price, error) {
+	tickerNew, err := ticker.GetTicker(f.Name, p, assetType)
+	if err != nil {
+		return f.UpdateTicker(p, assetType)
+	}
+	return tickerNew, nil
+}
+
+// FetchOrderbook returns orderbook base on the currency pair
+func (f *FTX) FetchOrderbook(currency currency.Pair, assetType asset.Item) (*orderbook.Base, error) {
+	ob, err := orderbook.Get(f.Name, currency, assetType)
+	if err != nil {
+		return f.UpdateOrderbook(currency, assetType)
+	}
+	return ob, nil
+}
+
+// UpdateOrderbook updates and returns the orderbook for a currency pair
+func (f *FTX) UpdateOrderbook(p currency.Pair, assetType asset.Item) (*orderbook.Base, error) {
+	orderBook := new(orderbook.Base)
+	tempResp, err := f.GetOrderbook(f.FormatExchangeCurrency(p, assetType).String(), 0)
+	if err != nil {
+		return orderBook, err
+	}
+	for x := range tempResp.Bids {
+		orderBook.Bids = append(orderBook.Bids, orderbook.Item{
+			Amount: tempResp.Bids[x].Size,
+			Price:  tempResp.Bids[x].Price})
+	}
+	for y := range tempResp.Asks {
+		orderBook.Asks = append(orderBook.Asks, orderbook.Item{
+			Amount: tempResp.Asks[y].Size,
+			Price:  tempResp.Asks[y].Price})
+	}
+	orderBook.Pair = p
+	orderBook.ExchangeName = f.Name
+	orderBook.AssetType = assetType
+	err = orderBook.Process()
+	if err != nil {
+		return orderBook, err
+	}
+	return orderbook.Get(f.Name, p, assetType)
+}
+
+// UpdateAccountInfo retrieves balances for all enabled currencies
+func (f *FTX) UpdateAccountInfo() (account.Holdings, error) {
+	var resp account.Holdings
+	data, err := f.GetBalances()
+	if err != nil {
+		return resp, err
+	}
+	var acc account.SubAccount
+	for key := range data.Result {
+		c := currency.NewCode(data.Result[key].Coin)
+		hold := data.Result[key].Total - data.Result[key].Free
+		total := data.Result[key].Total
+		acc.Currencies = append(acc.Currencies,
+			account.Balance{CurrencyName: c,
+				TotalValue: total,
+				Hold:       hold})
+	}
+	resp.Accounts = append(resp.Accounts, acc)
+	resp.Exchange = f.Name
+
+	err = account.Process(&resp)
+	if err != nil {
+		return account.Holdings{}, err
+	}
+
+	return resp, nil
+}
+
+// FetchAccountInfo retrieves balances for all enabled currencies
+func (f *FTX) FetchAccountInfo() (account.Holdings, error) {
+	acc, err := account.GetHoldings(f.Name)
+	if err != nil {
+		return f.UpdateAccountInfo()
+	}
+
+	return acc, nil
+}
+
+// GetFundingHistory returns funding history, deposits and
+// withdrawals
+func (f *FTX) GetFundingHistory() ([]exchange.FundHistory, error) {
+	return nil, common.ErrNotYetImplemented
```

Comment:
Please implement this https://docs.ftx.com/#get-deposit-history | https://docs.ftx.com/#get-withdrawal-history